core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeValueHandler.java

```diff
@@ -0,0 +1,82 @@
+package org.hisp.dhis.android.core.trackedentity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackedEntityAttributeValueHandler {
+    private final TrackedEntityAttributeValueStore trackedEntityAttributeValueStore;
+
+    public TrackedEntityAttributeValueHandler(
+            TrackedEntityAttributeValueStore trackedEntityAttributeValueStore) {
+        this.trackedEntityAttributeValueStore = trackedEntityAttributeValueStore;
+    }
+
+    public void handle(String trackedEntityInstanceUid,
+            List<TrackedEntityAttributeValue> attributeValues) {
+        if (trackedEntityInstanceUid == null || attributeValues == null) {
+            return;
+        }
+
+        removeNoExistedAttributeValuesInServer(trackedEntityInstanceUid, attributeValues);
+
+        for (TrackedEntityAttributeValue attValue : attributeValues) {
+            handle(trackedEntityInstanceUid, attValue);
+        }
+    }
+
+    private void removeNoExistedAttributeValuesInServer(String trackedEntityInstanceUid,
+            List<TrackedEntityAttributeValue> attributeValues) {
+
+        List<String> uIds = getAttributeUIdsToRemove(trackedEntityInstanceUid, attributeValues);
+
+        if (!uIds.isEmpty()) {
+            trackedEntityAttributeValueStore.deleteByInstanceAndAttributes(
+                    trackedEntityInstanceUid, uIds);
+        }
+    }
+
+    private List<String> getAttributeUIdsToRemove(String trackedEntityInstanceUid,
+            List<TrackedEntityAttributeValue> attributeValues) {
+        List<String> attributeUIdsToRemove = new ArrayList<>();
+
+        List<TrackedEntityAttributeValue> attributeValuesInDB =
+                trackedEntityAttributeValueStore.queryByTrackedEntityInstance(
+                        trackedEntityInstanceUid);
+
+        for (TrackedEntityAttributeValue attributeValue : attributeValuesInDB) {
+            if (!existsDataElement(attributeValues, attributeValue.trackedEntityAttribute())) {
+                attributeUIdsToRemove.add(attributeValue.trackedEntityAttribute());
+            }
+        }
+
+        return attributeUIdsToRemove;
+    }
+
+    private boolean existsDataElement(List<TrackedEntityAttributeValue> attributeValues,
```

Comment:
@xurxodev didn't you mean something like existsTrackedEntityAttribute here? Looking at the code, it looks like it's checking if the TE attributes exist for a list of TE attributeValues. Is that it?

Reply:
@ifoche yes sorry, It's a copy and paste from events solution, I have renamed these method